crates/plugins/src/plugin_manifest.rs

```diff
@@ -0,0 +1,168 @@
+use serde::{Deserialize, Serialize};
+
+/// Expected schema of a plugin manifest. Should match the latest Spin plugin
+/// manifest JSON schema:
```

Comment:
We went with JSON because of the large ecosystem around JSON schema documentation and JSON parsing for schema matching. However, I did not realize that TOML also can be [parsed with JSON schema](https://json-schema-everywhere.github.io/toml); though, I cannot see examples. Have you seen examples of TOML schemas?

Reply:
The "Even Better TOML" VS Code extension does some stuff with TOML schemas.  But yeah, it's not as widely used.  It's a definite advantage of JSON.

Should we migrate to JSON for our other manifests to take advantage of schema support, or does the plugin manifest have specific considerations (e.g. higher complexity, less human authoring) that make schema support more important (or human readability less important) than in other manifests?

Reply:
I like how JSON schemas enable specifying field options (ie OS enum), verify names (regex and enums), can specify manditory and optional fields, and act as documentation on how to construct a manifest. However, the Spin manifest tomls are refreshingly easy to read. I wonder if we can make manifests TOML but when validating them, convert them to JSON and check them against a schema. That could give us the best of both worlds. (Unfortunately the TOML schema parser is unmaintained: https://github.com/json-schema-everywhere/pajv, which is which the initial convert to JSON step may be needed)

Reply:
I played around with this in this [branch of `spin-plugins`](https://github.com/kate-goldenring/spin-plugins/blob/toml-manifests/README.md#validating-plugin-schemas)

Reply:
It feels a little messy. In the end, @itowlson, I think your note summarizes it: spin plugin manifests are less frequently authored and can use the benefits of json tooling (schema validation). However, spin app manifests are validated when they are run, so they dont need json schema validation. So i think using json for spin plugin manifests and toml for app manifests is appropriate